chains/evm/contracts/test/feeQuoter/FeeQuoter.parseSuiExtraArgsFromBytes.t.sol

```diff
@@ -0,0 +1,94 @@
+
+  function setUp() public virtual override {
+    super.setUp();
+    s_destChainConfig = _generateFeeQuoterDestChainConfigArgs()[0].destChainConfig;
+    s_destChainConfig.enforceOutOfOrder = true; // Enforcing out of order execution for messages to SVM
+    s_destChainConfig.chainFamilySelector = Internal.CHAIN_FAMILY_SELECTOR_SUI;
+
+    FeeQuoter.DestChainConfigArgs[] memory destChainConfigs = new FeeQuoter.DestChainConfigArgs[](1);
+    destChainConfigs[0] =
+      FeeQuoter.DestChainConfigArgs({destChainSelector: DEST_CHAIN_SELECTOR, destChainConfig: s_destChainConfig});
+    s_feeQuoter.applyDestChainConfigUpdates(destChainConfigs);
+  }
+
+  function test_SuiExtraArgsV1TagSelector() public pure {
+    assertEq(Client.SUI_EXTRA_ARGS_V1_TAG, bytes4(keccak256("CCIP SuiExtraArgsV1")));
+  }
+
+  function test_SuiExtraArgsV1() public view {
+    Client.SuiExtraArgsV1 memory inputArgs = Client.SuiExtraArgsV1({
+      gasLimit: GAS_LIMIT,
+      allowOutOfOrderExecution: true,
+      tokenReceiver: bytes32(uint256(0)),
+      receiverObjectIds: new bytes32[](2)
+    });
+
+    bytes memory inputExtraArgs = Client._suiArgsToBytes(inputArgs);
+
+    Client.SuiExtraArgsV1 memory expectedOutputArgs = Client.SuiExtraArgsV1({
+      gasLimit: GAS_LIMIT,
+      allowOutOfOrderExecution: true,
+      tokenReceiver: bytes32(uint256(0)),
+      receiverObjectIds: new bytes32[](2)
+    });
+
+    vm.assertEq(
+      abi.encode(s_feeQuoter.parseSuiExtraArgsFromBytes(inputExtraArgs, s_destChainConfig)),
+      abi.encode(expectedOutputArgs)
+    );
+  }
+
+  // Reverts
+  function test_RevertWhen_ExtraArgsAreEmpty() public {
+    bytes memory inputExtraArgs = new bytes(0);
+    vm.expectRevert(FeeQuoter.InvalidExtraArgsData.selector);
+    s_feeQuoter.parseSuiExtraArgsFromBytes(inputExtraArgs, s_destChainConfig);
+  }
+
+  function test_RevertWhen_InvalidExtraArgsTag() public {
+    bytes memory inputExtraArgs = abi.encodeWithSelector(bytes4(0));
+
+    vm.expectRevert(FeeQuoter.InvalidExtraArgsTag.selector);
+    s_feeQuoter.parseSuiExtraArgsFromBytes(inputExtraArgs, s_destChainConfig);
+  }
+
+  function test_RevertWhen_SuiMessageGasLimitTooHigh() public {
+    Client.SuiExtraArgsV1 memory inputArgs = Client.SuiExtraArgsV1({
+      gasLimit: s_destChainConfig.maxPerMsgGasLimit + 1,
+      allowOutOfOrderExecution: true,
+      tokenReceiver: bytes32(uint256(0)),
+      receiverObjectIds: new bytes32[](2)
+    });
+
+    bytes memory inputExtraArgs = Client._suiArgsToBytes(inputArgs);
+
+    vm.expectRevert(FeeQuoter.MessageGasLimitTooHigh.selector);
+    s_feeQuoter.parseSuiExtraArgsFromBytes(inputExtraArgs, s_destChainConfig);
+  }
+
+  function test_RevertWhen_ExtraArgOutOfOrderExecutionIsFalse() public {
+    bytes memory inputExtraArgs = abi.encodeWithSelector(
+      Client.SUI_EXTRA_ARGS_V1_TAG,
+      Client.SuiExtraArgsV1({
+        gasLimit: GAS_LIMIT,
+        allowOutOfOrderExecution: false,
+        tokenReceiver: bytes32(uint256(0)),
```

Comment:
tokenReceiver is zero in every single test, it would probably make sense to have it be non-zero somewhere